projects/self_feeding/self_feeding_agent.py

```diff
@@ -302,65 +302,83 @@ def act(self):
                 rating = self.extract_rating()
                 if rating == -1:
                     action = self.make_action(
-                        self.make_explanation_request(), reward=rating
+                        self.make_feedback_request(), reward=rating
                     )
-                    self.status = EXPLANATION_REQUESTED
+                    self.status = FEEDBACK_REQUESTED
 
                 else:
                     action, reply = self.make_rating_response(rating)
                     self.status = RATING_ACCEPTED
                     # Store just the non-template part of the response in the history
-                    self.history.append(reply)
+                    self.history.update_history({"text": reply})
 
-            elif self.status == EXPLANATION_REQUESTED:
+            elif self.status == FEEDBACK_REQUESTED:
                 action = self.make_action(THANKS + ' ' + NEWTOPIC)
                 self.status = NEWTOPIC_REQUESTED
-                self.history.clear()
+                self.history.reset()
 
             else:
                 raise Exception(f"Unrecognized status: {self.status}")
 
         if self.status == NORMAL:
             action = super().act()
-            self.history.append(action['text'])
+            self.history.update_history(action)
 
         return action
 
     def observe(self, observation):
         """Add to history, concatenate history-size utterances, and add person tokens"""
+
         # If their response is a response to a rating request, no work required
         if self.status == RATING_REQUESTED:
             self.last_rating = observation['text']
         else:
             if 'text' in observation:
```

Comment:
you don't need this check here, calling `update_history` will check `if 'text' in observation`